### PR TITLE
collision control wall

### PR DESCRIPTION
**Pull request by terminalobject**

### CODE REVIEW

File: src/ball.js [hunk 1]
```diff
@@ -24,12 +24,13 @@
...
-  Ball.prototype.directionUpdate = function () {
-    if (this.y + this.dy - this.radius < 0) {
-      this.dy = -this.dy
-    } else if (this.y + this.dy + this.radius > this.canvas.height) {
-      this.dy = -this.dy
+  Ball.prototype.bounce = function (surface) {
+    if (surface = "wall") {
+      this.dy = -this.dy;
     }
+  //  else if (surface = "paddle") {
```
Comment (tbscanlon) [inline src/ball.js:31]: Will this be added later to implement the ball bouncing off the paddle?

File: src/game.js [hunk 2]
```diff
@@ -36,6 +36,15 @@
     this.ball.draw(this.ball.x, this.ball.y);
   };
 
+  Game.prototype.collisionDetection = function () {
+    if (this.ball.y + this.ball.dy - this.ball.radius < 0) {
```
Comment (tbscanlon) [inline src/game.js]: can the contents of the if clause be moved into a private method that returns true or false? 
e.g. `if ( ballWillBounce() )`

File: src/game.js [hunk 3]
```diff
@@ -36,6 +36,15 @@
...
+      this.ball.bounce("wall");
+    } 
+    else if (this.ball.y + this.ball.dy + this.ball.radius > this.canvas.height) {
```
Comment (tbscanlon) [inline src/game.js]: As above - our concern is that the code is less readable with long conditionals

**Review by tbscanlon — CHANGES_REQUESTED** [review]

File: src/ball.js (hunk 1 above)
Comment (alistairkung) [reply src/ball.js:31]: yes

**Review by tbscanlon — APPROVED** [review]
:shipit: :night_with_stars: :tanabata_tree:

**Pull request merged**